### DISCUSSION TITLE
error: Found argument '--allow-plugin' which wasn't expected, or isn't valid in this context Did you mean --allow-run?

### DISCUSSION THREAD

**Discussion by goodmanworld** [denoland/deno#12365] [Q&A] [answered]
I am a new Deno learner and am trying to following a web tutorial about it. But when I try to run my application with the following command, it gives me the followed error:

    deno run --allow-net --allow-write --allow-read --allow-plugin --allow-env --unstable server.ts server.ts

> error: Found argument '--allow-plugin' which wasn't expected, or isn't
> valid in this context. Did you mean --allow-run?

**Comment by kitsonk** [comment, marked as answer]
The tutorial you are using is outdated for the version of Deno you are using.  Unstable plugins were removed from the CLI.  I would contact the tutorial author and let them know it is outdated.

**Reply by goodmanworld** [reply]
> 
> 
> The tutorial you are using is outdated for the version of Deno you are using. Unstable plugins were removed from the CLI. I would contact the tutorial author and let them know it is outdated.

Do you know what tutorial I am using? You say I must wait until the problems resolves automatically or must to some alternative?

**Reply by kitsonk** [reply]
I don't need to know what tutorial you are using.  `--allow-plugin` was removed from Deno.  It was an unstable API and replaced with FFI.  Therefore any tutorial instructing you to use it is out of date.  It will not resolve itself, because the functionality was removed.

**Comment by cs09g** [comment]
"Installing from deno.land"($ deno install -qAn vr https://deno.land/x/velociraptor@1.3.0/cli.ts) causes `--allow-plugin` issue.

It writes below on vr path(`/Users/user/.deno/bin/vr`). There's still `--allow-plugin`.
`exec deno run --allow-read --allow-write --allow-net --allow-env --allow-run --allow-plugin --allow-hrtime --quiet 'https://deno.land/x/velociraptor@1.3.0/cli.ts' "$@"`

"installing from nest.land" is fine.

**Reply by umbopepato** [reply]
@cs09g please see https://github.com/jurassiscripts/velociraptor/issues/75#issuecomment-909071188
